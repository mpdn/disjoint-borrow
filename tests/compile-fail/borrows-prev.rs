extern crate disjoint_borrow;
use disjoint_borrow::DisjointSlice;

fn main() {
    let mut a = [1, 2, 3];
    let mut alpha = DisjointSlice::new(&mut a);
    let (beta, b) = alpha.get(1..2);
    drop(beta);
    alpha.get(1..2);
    //~^ cannot borrow `alpha` as mutable more than once at a time
    b[0];
}