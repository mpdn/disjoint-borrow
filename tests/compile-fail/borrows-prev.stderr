error[E0499]: cannot borrow `alpha` as mutable more than once at a time
  --> $DIR/borrows-prev.rs:9:5
   |
7  |     let (beta, b) = alpha.get(1..2);
   |                     ----- first mutable borrow occurs here
8  |     drop(beta);
9  |     alpha.get(1..2);
   |     ^^^^^ second mutable borrow occurs here
10 |     b[0];
   |     ---- first borrow later used here

For more information about this error, try `rustc --explain E0499`.
